### PR TITLE
e2e: remove opt-in skip in the node-selector workload placement test

### PR DESCRIPTION
**Pull request by fromanirh**
Add the fixes which allow us to always run this test, so removing the opt-in skip.
See individual patches for details.

### CODE REVIEW

**Comment by fromanirh** [comment]
funny, the linter passes just fine locally:
```
$ golangci-lint run --out-format=github-actions --timeout=15m0s --verbose
INFO [config_reader] Config search paths: [./ /home/fromani/go/src/github.com/openshift-kni/numaresources-operator /home/fromani/go/src/github.com/openshift-kni /home/fromani/go/src/github.com /home/fromani/go/src /home/fromani/go /home/fromani /home /] 
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck] 
INFO [loader] Go packages loading at mode 575 (types_sizes|name|deps|exports_file|files|imports|compiled_files) took 1.051624505s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 12.87746ms 
INFO [linters context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Issues before processing: 3, after processing: 0 
INFO [runner] Processors filtering stat (out/in): cgo: 3/3, path_prettifier: 3/3, filename_unadjuster: 3/3, skip_files: 3/3, autogenerated_exclude: 3/3, identifier_marker: 3/3, exclude-rules: 0/3, exclude: 3/3, skip_dirs: 3/3 
INFO [runner] processing took 232.369µs with stages: path_prettifier: 97.75µs, autogenerated_exclude: 56.384µs, identifier_marker: 43.734µs, skip_dirs: 17.207µs, exclude-rules: 11.693µs, cgo: 1.151µs, nolint: 1.1µs, filename_unadjuster: 655ns, max_same_issues: 465ns, uniq_by_line: 441ns, max_from_linter: 257ns, diff: 248ns, source_code: 224ns, exclude: 222ns, skip_files: 198ns, severity-rules: 187ns, path_shortener: 126ns, sort_results: 118ns, max_per_file_from_linter: 113ns, path_prefixer: 96ns 
INFO [runner] linters took 295.196951ms with stages: goanalysis_metalinter: 294.853493ms 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 15 samples, avg is 80.5MB, max is 138.6MB 
INFO Execution took 1.365171223s                  

$ golangci-lint --version
golangci-lint has version 1.41.1 built from a2074809 on 2021-06-19T16:01:50Z
```

**Comment by fromanirh** [comment]
looks like we have a cache corruption issue in gh actions, because 
1. the error doesn't make much sense and
2. the same linter (and with the same version) works fine locally

**Comment by fromanirh** [comment]
https://github.com/golangci/golangci-lint-action#caching-internals
TL;DR:

- change the go.mod file (https://github.com/openshift-kni/numaresources-operator/pull/246) and/or
- let us just wait for few days

File: test/e2e/serial/tests/workload_placement_nodesel.go [hunk 1]
```diff
@@ -190,13 +199,12 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 			err = fxt.Client.Create(context.TODO(), pod)
 			Expect(err).NotTo(HaveOccurred(), "unable to create pod %q", pod.Name)
 
-			By("waiting for node to be up&running")
-			podRunningTimeout := 1 * time.Minute
-			updatedPod, err := e2ewait.ForPodPhase(fxt.Client, pod.Namespace, pod.Name, corev1.PodRunning, podRunningTimeout)
+			By("waiting for node to be up & running")
```
Comment (shajmakh) [inline test/e2e/serial/tests/workload_placement_nodesel.go:199]: "waiting for pod to be up .."

File: test/e2e/serial/tests/workload_placement_nodesel.go [hunk 2]
```diff
@@ -190,13 +199,12 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
...
+			updatedPod, err := e2ewait.ForPodPhase(fxt.Client, pod.Namespace, pod.Name, corev1.PodRunning, 1*time.Minute)
 			if err != nil {
 				_ = objects.LogEventsForPod(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name)
 			}
-			Expect(err).NotTo(HaveOccurred(), "Pod %q not up&running after %v", pod.Name, podRunningTimeout)
+			Expect(err).NotTo(HaveOccurred())
```
Comment (shajmakh) [inline test/e2e/serial/tests/workload_placement_nodesel.go:204]: why remove the note?

File: test/e2e/serial/tests/workload_placement_nodesel.go [hunk 3]
```diff
@@ -111,23 +115,23 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
...
-			Expect(ok).To(BeTrue(), "cannot select a targe node among %#v", nrtCandidateNames.List())
+			Expect(ok).To(BeTrue(), "cannot select a target node among %#v", nrtCandidateNames.List())
 			By(fmt.Sprintf("selecting node to schedule the pod: %q", toAlsoLabelNodeName))
 
 			labelName := "size"
 			labelValue := "medium"
 			By(fmt.Sprintf("Labeling nodes %q and %q with label %q:%q", targetNodeName, toAlsoLabelNodeName, labelName, labelValue))
 
-			unlabelOneFunc, err := labelNodeWithValue(fxt.Client, labelName, labelName, targetNodeName)
+			unlabelOneFunc, err := labelNodeWithValue(fxt.Client, labelName, labelValue, targetNodeName)
```
Comment (shajmakh) [inline test/e2e/serial/tests/workload_placement_nodesel.go:129]: there is similar wrong labeling for node `toAlsoLabelNodeName`

File: test/e2e/serial/tests/workload_placement_nodesel.go (hunk 2 above)
Comment (fromanirh) [reply test/e2e/serial/tests/workload_placement_nodesel.go:204]: I don't see a lot of value added, but I guess I can put it back

File: test/e2e/serial/tests/workload_placement_nodesel.go (hunk 1 above)
Comment (fromanirh) [reply test/e2e/serial/tests/workload_placement_nodesel.go:199]: right, fixing

File: test/e2e/serial/tests/workload_placement_nodesel.go (hunk 2 above)
Comment (shajmakh) [reply test/e2e/serial/tests/workload_placement_nodesel.go:204]: it's okay both ways, I think pod events will be sufficient anyway

File: test/e2e/serial/tests/workload_placement_nodesel.go (hunk 3 above)
Comment (fromanirh) [reply test/e2e/serial/tests/workload_placement_nodesel.go:129]: good catch, fixed!
